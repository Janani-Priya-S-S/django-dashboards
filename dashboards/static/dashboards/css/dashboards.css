:root {
  --color-background-primary: #e5e5e5;
  --color-background-secondary: #fafafa;
  --color-background-tertiary: #f5f5f5;
  --color-border: #d4d4d4;
  --color-text-primary: #404040;
  --color-text-secondary: #a3a3a3;
  --shadow: 0 0 0 0px #fff, 0 20px 25px -5px rgba(0,0,0,0.10196), 0 8px 10px -6px rgba(0,0,0,0.10196);
  --root-background: linear-gradient(to top right, var(--color-background-primary), var(--color-background-primary), var(--color-border));
  --color-danger: #dc2626;
  --color-cta-background: #737373;
  --color-cta-hover-background: #404040;
  --color-cta-text: #fafafa;
  --color-cta-hover-text: var(--color-cta-text);
  --color-cta-alt-background: #a3a3a3;
  --color-cta-alt-hover-background: #fafafa;
  --color-cta-alt-text: #404040;
  --color-cta-alt-hover-text: var(--color-cta-alt-text);
  --transition-time: 0.25s;
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-background-primary: #404040;
    --color-background-secondary: #525252;
    --color-background-tertiary: #737373;
    --color-border: #737373;
    --color-text-primary: #fafafa;
    --color-text-secondary: #a3a3a3;
    --root-background: linear-gradient(to top right, var(--color-background-primary), var(--color-background-primary), var(--color-background-secondary));
    --color-danger: #fca5a5;
    --color-cta-background: #a3a3a3;
    --color-cta-hover-background: #fafafa;
    --color-cta-text: #404040;
    --color-cta-hover-text: var(--color-cta-text);
    --color-cta-alt-background: #737373;
    --color-cta-alt-hover-background: #404040;
    --color-cta-alt-text: #fafafa;
    --color-cta-alt-hover-text: var(--color-cta-alt-text);
  }
}

.btn {
  height: 2em;
  border-radius: 1em;
  text-align: center;
  border: 0;
  box-shadow: var(--shadow);
  transition: all var(--transition-time);
  padding: 0 1em;
}

.btn.cta {
  background-color: var(--color-cta-background);
  color: var(--color-cta-text);
}

.btn.cta:hover,
.btn.cta:active {
  background-color: var(--color-cta-hover-background);
  color: var(--color-cta-hover-text);
}

.btn.cta.alt {
  background-color: var(--color-cta-alt-background);
  color: var(--color-cta-alt-text);
}

.btn.cta.alt:hover,
.btn.cta.alt:active {
  background-color: var(--color-cta-alt-hover-background);
  color: var(--color-cta-alt-hover-text);
}

.card {
  position: relative;
  height: 100%;
}

.card .card-header {
  display: flex;
  justify-content: space-between;
  margin-bottom: 0;
  background-color: var(--color-background-tertiary);
  padding: 1rem;
  border-bottom: solid 1px rgba(154, 161, 171, 0.15);
  border-bottom: solid 1px var(--color-border);
}

.card .card-header .header-title {
  flex: 1 1 auto;
  font-size: 1.2em;
}

.card .card-header .dropdown {
  position: relative;
  display: inline-block;
}

.card .card-header .dropdown .dropdown-btn {
  color: white;
  padding: 6px;
  font-size: 1em;
  border: none;
  cursor: pointer;
}

.card .card-header .dropdown .dropdown-content {
  position: absolute;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.card .card-header .dropdown .dropdown-content a {
  color: black;
  font-size: 1em;
  padding: 12px 16px;
  -webkit-text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.card .card-header .dropdown .dropdown-content a:hover {
  background-color: #f1f1f1;
}

.card .card-img {
  width: 100%;
}

.card .card-body {
  padding: 5px;
}

.card-body-grid {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-column-gap: 10px;
}

.card .card-body p {
  font-size: 1rem;
}

.card .card-footer {
  flex: 1 1 auto;
  padding: 5px;
  font-size: 0.75em;
  background-color: var(--color-background-tertiary);
}

.form-table {
  text-align: left;
}

.form-table label {
  font-weight: bold;
}

.form-table input,
.form-table textarea,
.form-table select {
  padding: 0.5em;
  border-radius: 0.5em;
  width: 100%;
  border: 1px solid var(--border-color);
  background: var(--color-background-primary);
  color: var(--color-text-primary)
}

.form-table .helptext {
  color: var(--color-text-secondary);
  margin: 0;
  font-size: 0.7rem;
}

.errorlist {
  color: var(--color-danger);
  list-style: none;
  text-align: left;
  padding-left: 10px;
  margin: 0;
}

.form-table .errorlist {
  padding-left: 0;
}

* {
  box-sizing: border-box;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-kerning: auto;
}

html {
  font-size: 1.2em;
  line-height: 1.4;
  font-weight: 400;
  font-family: Inter var,ui-sans-serif,system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;
  background: var(--root-background);
  color: var(--color-text-primary);
}

body {
  padding: 1em;
  margin: 0 auto;
  max-width: 95%;
  min-height: 100vh;
}

a{
  color: var(--color-text-primary);
  cursor: pointer;
  font-size: 1.1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

h1, h2, h3, h4, h5, h6 {
  margin: 0 0 .5em 0;
  line-height: 1.2;
  letter-spacing: -.02em;
}

hr{
  border-color: var(--color-text-primary);
}

.menu {
  background: var(--color-background-secondary);
  list-style: none;
  margin: 0;
  padding-left: 0;
  display: flex;
  z-index: 2;
}

.menu li {
  background: var(--color-background-secondary);
  display: block;
  float: left;
  padding: 0.5rem 2rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.menu li a {
  color: var(--color-text-primary);
}

.menu li.active, .menu li ul li.active {
  background-color: var(--color-background-tertiary);
}

.menu li:hover,
.menu li:focus-within {
  background-color: var(--color-background-tertiary);
  cursor: pointer;
}

.menu li:focus-within a {
  outline: none;
}

.menu .dropdown {
  display: none;
  left: 0;
  margin: 0.5rem 0 0;
  padding: 0;
  min-width: 4em;
  opacity: 0;
  position: absolute;
  visibility: hidden;
  background: var(--color-background-tertiary);
  border: 1px solid var(--color-border);
  border-top: 0;
  overflow: hidden;
  border-radius: 0 0 15px 15px;
}

.menu li:hover > .dropdown,
.menu li:focus-within > .dropdown,
.menu li .dropdown:hover,
.menu li .dropdown:focus {
  visibility: visible;
  opacity: 1;
  display: block;
  z-index: 1;
}

.menu .dropdown li {
  clear: both;
  width: 100%;
  padding: 0;
}

.menu .dropdown li a {
  display: block;
  padding: 0.5rem 2rem;
}

.menu-card {
  display: flex;
  margin-bottom: 1em;
}

.menu-card nav {
  flex-grow: 1;
  box-shadow: var(--shadow);
  border: 1px solid var(--color-border);
  border-right: 0;
  border-left: 0;
}

.menu-card .bumper {
  width: 15px;
  overflow: hidden;
  background: var(--color-background-secondary);
  border: 1px solid var(--color-border);
  z-index: 1;
}

.menu-card .bumper:first-child {
  border-radius: 15px 0 0 15px;
  border-right: 0;
}

.menu-card .bumper:last-child {
  border-radius: 0 15px 15px 0;
  border-left: 0;
}

/* This is an example - you'd ideally use a lib or scss to compile this, add responsiveness etc */

.span-1 {
  grid-column-end: span 1 !important;
}

.span-1 {
  grid-auto-rows: min-content;
}

.span-2 {
  grid-column-end: span 2 !important;
}

.span-2 {
  grid-auto-rows: min-content;
}

.span-3 {
  grid-column-end: span 3 !important;
}

.span-3 {
  grid-auto-rows: min-content;
}

.span-4 {
  grid-column-end: span 4 !important;
}

.span-4 {
  grid-auto-rows: min-content;
}

.span-5 {
  grid-column-end: span 5 !important;
}

.span-5 {
  grid-auto-rows: min-content;
}

.span-6 {
  grid-column-end: span 6;
  grid-auto-rows: min-content;
}

.span-7 {
  grid-column-end: span 7;
  grid-auto-rows: min-content;
}

.span-8 {
  grid-column-end: span 8;
  grid-auto-rows: min-content;
}

.span-9 {
  grid-column-end: span 9;
  grid-auto-rows: min-content;
}

.span-10 {
  grid-column-end: span 10;
  grid-auto-rows: min-content;
}

.span-11 {
  grid-column-end: span 11;
  grid-auto-rows: min-content;
}

.span-12 {
  grid-column-end: span 12;
  grid-auto-rows: min-content;
}

.dashboard-container {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-gap: 1rem;
  gap: 1rem;
  text-align: center;
  width: 100%;
}

h1, h2, h3, h4, p, .tab-container {
  grid-column: 1 / 13;
}

.dashboard-component {
  border-radius: 15px;
  background-color: var(--color-background-secondary);
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  box-shadow: var(--shadow);
  border: 1px solid var(--color-border);
}

.dashboard-component * {
  /* allow sub nesting, by assuming not set is 12 */
  grid-column-end: span 12;
}

.tab-content div.active .dashboard-component {
  border-radius: 0 15px 15px 15px;
}

.dashboard-component-inner {
  padding: 2rem;
}

.dashboard-component-cta {
  color: var(--color-text-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.3rem;
}

.tabs {
  border-top: 2px var(--color-border) solid;
  display: flex;
  flex-wrap: wrap;
  margin-left: 0;
  padding: 35px 0 0;
  list-style: none;
}

.tabs div, .tabs .tab {
  background-color: var(--color-background-secondary);
  padding: 10px 20px;
}

.tabs div a, .tabs .tab a {
  color: var(--color-text-primary);
}

.tabs div a.active, .tabs .tab a.active {
  color: var(--color-text-primary);
  border-bottom: 1px solid var(--color-background-secondary);
}

.dataTables_filter, .dataTables_length, .dataTables_info, .dataTables_paginate {
  color: var(--color-text-primary) !important;
  font-size: 0.9rem !important;
}

.dataTables_filter, .dataTables_length, .dataTables_info, .dataTables_paginate {
  margin-bottom: 10px;
}

.dataTables_length select {
  color: var(--color-text-primary) !important;
}

.table {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  font-size: 0.8rem;
}

.table thead th {
  background-color: var(--color-background-secondary);
  padding: 1rem;
  text-align: left;
  text-shadow: 1px 1px 1px var(--color-border);
}

.table tbody td {
  border: solid 1px var(--color-background-secondary);
  padding: 10px;
}

.customComponent {
  background-color: var(--color-background-primary);
  border-radius: 20px;
  padding: 10px 5px;
  margin: 10px;
  font-size: 0.8em;
}

.gauge-container {
  margin: auto;
  max-width: 65%;
}
