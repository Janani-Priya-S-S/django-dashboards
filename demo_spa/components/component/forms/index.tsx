import React, {useContext} from "react";
import {
    Value,
} from "@/types";
import {FilterContext} from "../../../appContext";

type Field = {
    name: string
    label: string
    value: string
    help_text: string
    id: string
    field_type: string
    required: boolean
    choices: string[]
}

type FormFieldComponentProps = {
    element: Field
    onChange: any
}

const FormFieldComponent = ({element, onChange} : FormFieldComponentProps) => {
    if (element.field_type === 'Select') {
        const choices = element.choices.map((choice) => <option value={choice[0]}>{choice[1]}</option>)
        return <select id={element.id} name={element.name} value={element.value} onChange={onChange}>
            {choices}
        </select>

    } else if (element.field_type === 'Textarea') {
        return <textarea name={element.name} id={element.id} onChange={onChange}>{element.value}</textarea>
    } else {
        return <input type={"text"} id={element.id} name={element.name} value={element.value} onChange={onChange} />
    }
}

export const Form = ({value}: { value: Value }) => {
    const [filter, setFilter] = useContext(FilterContext)
    const data = JSON.parse(JSON.stringify(value))
    const method = data.method

    const onChange = event => {
        setFilter(filter => ({ ...filter, [event.target.name]: event.target.value }))
    }

    const onSubmit = event => {
        event.preventDefault()
    }

    return (
        <form method={method} onSubmit={onSubmit}>
            {data.form.map((field: Field) => (
                <p key={field.id}>
                    <label>
                        {field.label}
                        <FormFieldComponent element={field} onChange={onChange}/>
                        {field.help_text != "" && <div>{field.help_text}</div>}
                    </label>
                </p>
            ))}
        </form>
    )
}
