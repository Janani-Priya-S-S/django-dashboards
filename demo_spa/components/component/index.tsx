export * from "./charts"
export * from "./html"
export * from "./table"
export * from "./text"
export * from "./display"