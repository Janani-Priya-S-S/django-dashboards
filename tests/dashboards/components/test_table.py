from unittest.mock import patch

from django.contrib.auth.models import Permission, User
from django.contrib.contenttypes.models import ContentType
from django.forms import model_to_dict
from django.template import Context

import pytest

from tests.dashboards.fakes import fake_user
from tests.utils import render_component_test
from wildcoeus.dashboards.component import BasicTable, Table
from wildcoeus.dashboards.component.table import (
    TableData,
    TableFilter,
    TableQuerysetFilter,
    TableQuerysetSort,
    TableSerializer,
    TableSort,
)


pytest_plugins = [
    "tests.dashboards.fixtures",
]


@pytest.mark.parametrize("component_class", [Table, BasicTable])
@pytest.mark.parametrize(
    "component_kwargs",
    [
        {"value": "value"},
        {"defer": lambda **kwargs: "value"},
        {"value": "value", "css_classes": ["a", "b"]},
    ],
)
@pytest.mark.parametrize("htmx", [True, False])
def test_render(component_class, dashboard, component_kwargs, htmx, rf, snapshot):
    component = component_class(**component_kwargs)
    component.dashboard = dashboard
    component.key = "test"
    context = Context(
        {
            "component": component,
            "request": rf.get("/"),
        }
    )

    snapshot.assert_match(render_component_test(context, htmx=htmx))


@pytest.mark.django_db
@pytest.mark.parametrize("filter_class", [TableFilter, TableQuerysetFilter])
def test_filter__table_queryset__global__one_field(filter_class, dashboard):
    abc = fake_user(username="abc")
    fake_user(username="xyz")

    data = User.objects.all()
    expected = [abc]
    if filter_class == TableFilter:
        # as if a list of data instead
        data = map(model_to_dict, data)
        expected = list(map(model_to_dict, expected))

    result = filter_class(filters={"search[value]": "abc"}, fields=["username"]).filter(
        data
    )

    assert list(result) == expected


@pytest.mark.django_db
@pytest.mark.parametrize("filter_class", [TableFilter, TableQuerysetFilter])
def test_filter__table_queryset__global__two_field(filter_class, dashboard):
    abc = fake_user(username="abc")
    xyz = fake_user(username="xyz", first_name="abc")

    data = User.objects.all()
    expected = [abc, xyz]
    if filter_class == TableFilter:
        # as if a list of data instead
        data = map(model_to_dict, data)
        expected = list(map(model_to_dict, expected))

    result = filter_class(
        filters={"search[value]": "abc"}, fields=["username", "first_name"]
    ).filter(data)

    assert list(result) == expected


@pytest.mark.django_db
@pytest.mark.parametrize("filter_class", [TableFilter, TableQuerysetFilter])
def test_filter__table_queryset__individual__find_on_username(filter_class, dashboard):
    abc = fake_user(username="abc")
    fake_user(username="xyz", first_name="abc")

    data = User.objects.all()
    expected = [abc]
    if filter_class == TableFilter:
        # as if a list of data instead
        data = map(model_to_dict, data)
        expected = list(map(model_to_dict, expected))

    result = filter_class(
        filters={"columns[0][search][value]": "abc"}, fields=["username", "first_name"]
    ).filter(data)

    assert list(result) == expected


@pytest.mark.django_db
@pytest.mark.parametrize("filter_class", [TableFilter, TableQuerysetFilter])
def test_filter__table_queryset__individual__find_on_first_name(
    filter_class, dashboard
):
    xyz = fake_user(username="xyz", first_name="abc")

    data = User.objects.all()
    expected = [xyz]
    if filter_class == TableFilter:
        # as if a list of data instead
        data = map(model_to_dict, data)
        expected = list(map(model_to_dict, expected))

    result = filter_class(
        filters={"columns[1][search][value]": "abc"}, fields=["username", "first_name"]
    ).filter(data)

    assert list(result) == expected


@pytest.mark.django_db
@pytest.mark.parametrize(
    "filters,force_lower,expected_order",
    [
        ({}, True, [0, 1, 2]),
        ({"order[0][column]": 0}, True, [0, 2, 1]),
        ({"order[0][column]": 1}, True, [2, 0, 1]),
        (
            {"order[0][column]": 0, "order[0][dir]": "desc"},
            True,
            [1, 2, 0],
        ),
        # Not forced to lower
        ({}, False, [0, 1, 2]),  # Does not change
        ({"order[0][column]": 0}, False, [2, 0, 1]),
        ({"order[0][column]": 1}, False, [2, 0, 1]),  # Does not change
        (
            {"order[0][column]": 0, "order[0][dir]": "desc"},
            False,
            [1, 0, 2],
        ),
    ],
)
@pytest.mark.parametrize("sort_class", [TableSort, TableQuerysetSort])
def test_sort__table_queryset(sort_class, filters, force_lower, expected_order):
    one = fake_user(username="abc", first_name="123")
    two = fake_user(username="xyz", first_name="345")
    three = fake_user(username="CBA", first_name="012")

    data = User.objects.all()
    users = [one, two, three]

    if sort_class == TableSort:
        # as if a list of data instead
        data = map(model_to_dict, data)
        users = list(map(model_to_dict, [one, two, three]))

    result = sort_class(
        filters=filters, fields=["username", "first_name"], force_lower=force_lower
    ).sort(data)

    assert list(result) == [users[i] for i in expected_order]


@pytest.mark.django_db
@pytest.mark.parametrize("length", [5, 10, -1])
def test_serializer__queryset(length):
    for u in range(0, 11):
        fake_user()

    data = User.objects.all()
    result = TableSerializer(
        data=data,
        fields=["username", "first_name", "is_staff", "last_login", "date_joined"],
        count_func=lambda x: x.count(),
        filters={},
    ).get(start=0, length=length)

    expected_length = length if length > 0 else data.count()
    assert isinstance(result, TableData)
    assert len(result.data) == expected_length
    assert result.draw == 1
    assert result.total == 11
    assert result.filtered == 11


@pytest.mark.django_db
@pytest.mark.parametrize("length", [5, 10, -1])
def test_serializer__list(length):
    for u in range(0, 11):
        fake_user()

    data = list(User.objects.all())
    result = TableSerializer(
        data=data,
        fields=["username", "first_name", "is_staff", "last_login", "date_joined"],
        count_func=lambda x: len(x),
        filters={},
    ).get(start=0, length=length)

    expected_length = length if length > 0 else len(data)
    assert isinstance(result, TableData)
    assert len(result.data) == expected_length
    assert result.draw == 1
    assert result.total == 11
    assert result.filtered == 11


@pytest.mark.django_db
def test_serializer__sort_and_filter_applied():
    for u in range(0, 11):
        fake_user(username=str(u))

    data = User.objects.all()

    with patch.object(TableSort, "sort") as mock_sort:
        with patch.object(TableFilter, "filter") as mock_filter:
            result = TableSerializer(
                data=data,
                fields=[
                    "username",
                    "first_name",
                    "is_staff",
                    "last_login",
                    "date_joined",
                ],
                count_func=lambda x: x.count(),
                filters={"order[0][column]": 0, "order[0][dir]": "desc"},
                sort_class=TableSort,
                filter_class=TableFilter,
            ).get(start=0, length=5)

    assert isinstance(result, TableData)
    assert mock_sort.call_count == 1
    assert mock_filter.call_count == 1


@pytest.mark.django_db
def test_serializer__related_field():
    Permission.objects.create(
        name="Test",
        codename="test",
        content_type=ContentType.objects.get_for_model(User),
    )

    data = Permission.objects.filter(codename="test")
    result = TableSerializer(
        data=data,
        fields=["name", "content_type__name"],
        count_func=lambda x: x.count(),
        filters={},
    ).get(start=0, length=5)

    assert isinstance(result, TableData)
    assert result.data[0]["content_type__name"] == "user"


@pytest.mark.django_db
def test_serializer__invalid_fields():
    data = User.objects.all()
    result = TableSerializer(
        data=data,
        fields=["x", "y"],
        count_func=lambda x: x.count(),
        filters={},
    ).get(start=0, length=5)

    assert isinstance(result, TableData)
    assert len(result.data) == 0


@pytest.mark.django_db
def test_serializer__first_as_absolute_url():
    fake_user(username="abc", first_name="one")
    fake_user(username="def", first_name="two")

    class ProxyUser(User):
        def get_absolute_url(self):
            return f"/test/{self.username}"

        class Meta:
            app_label = "test"
            proxy = True

    data = ProxyUser.objects.all()
    result = TableSerializer(
        data=data,
        fields=["username", "first_name"],
        count_func=lambda x: x.count(),
        filters={},
        first_as_absolute_url=True,
    ).get(start=0, length=5)

    assert isinstance(result, TableData)
    assert result.data == [
        {"username": '<a href="/test/abc">abc</a>', "first_name": "one"},
        {"username": '<a href="/test/def">def</a>', "first_name": "two"},
    ]
