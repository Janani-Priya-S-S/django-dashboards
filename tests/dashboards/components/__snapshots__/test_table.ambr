# serializer version: 1
# name: test_render[False-component_kwargs0-BasicTable]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  
  <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%">
      <thead>
      
      </thead>
      <tbody>
      
      </tbody>
    </table>
  </div>
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[False-component_kwargs0-Table]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%"></table>
  </div>
  
  
    
      <script id="data_dashapp1testdashboardcomponenttest" type="application/json">"value"</script>
    
    <script id="columns_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script id="order_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script type="module">
        var columns_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('columns_dashapp1testdashboardcomponenttest').textContent);
        var order_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('order_dashapp1testdashboardcomponenttest').textContent);
  
        
            let data_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('data_dashapp1testdashboardcomponenttest').textContent);
            let rows_dashapp1testdashboardcomponenttest = data_dashapp1testdashboardcomponenttest.data;
  
            var options = {
                data: rows_dashapp1testdashboardcomponenttest,
                columns: columns_dashapp1testdashboardcomponenttest,
                pageLength: 10,
                scrollX: true,
                searching: true,
                paging: true,
                info: true,
                ordering: true,
                order: order_dashapp1testdashboardcomponenttest,
            }
  
        
  
        var table_dashapp1testdashboardcomponenttest = $('#dashapp1testdashboardcomponenttest_table').DataTable(options);
    </script>
  
  
  
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[False-component_kwargs1-BasicTable]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  
  <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%">
      <thead>
      
      </thead>
      <tbody>
      
      </tbody>
    </table>
  </div>
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[False-component_kwargs1-Table]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%"></table>
  </div>
  
  
    
    <script id="columns_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script id="order_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script type="module">
        var columns_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('columns_dashapp1testdashboardcomponenttest').textContent);
        var order_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('order_dashapp1testdashboardcomponenttest').textContent);
  
        
            $.ajaxSetup({
               headers: { "X-CSRFToken": JSON.parse(document.body.getAttribute("hx-headers"))["X-CSRFToken"]}
            });
            var options = {
                destroy: true,
                scrollX: true,
                processing: true,
                serverSide: true,
                searching: true,
                paging: true,
                info: true,
                ordering: true,
                columns: columns_dashapp1testdashboardcomponenttest,
                pageLength: 10,
                order: order_dashapp1testdashboardcomponenttest,
                ajax: {
                    url: "/dash/app1/testdashboard/@component/test/",
                    type: "POST",
                    dataFilter: function(data){
                      let json = jQuery.parseJSON( data );
                      json.recordsTotal = json.total;
                      json.recordsFiltered = json.filtered;
                      return JSON.stringify( json );
                  }
                }
            }
        
  
        var table_dashapp1testdashboardcomponenttest = $('#dashapp1testdashboardcomponenttest_table').DataTable(options);
    </script>
  
  
  
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[False-component_kwargs2-BasicTable]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  
  <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%">
      <thead>
      
      </thead>
      <tbody>
      
      </tbody>
    </table>
  </div>
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[False-component_kwargs2-Table]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%"></table>
  </div>
  
  
    
      <script id="data_dashapp1testdashboardcomponenttest" type="application/json">"value"</script>
    
    <script id="columns_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script id="order_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script type="module">
        var columns_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('columns_dashapp1testdashboardcomponenttest').textContent);
        var order_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('order_dashapp1testdashboardcomponenttest').textContent);
  
        
            let data_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('data_dashapp1testdashboardcomponenttest').textContent);
            let rows_dashapp1testdashboardcomponenttest = data_dashapp1testdashboardcomponenttest.data;
  
            var options = {
                data: rows_dashapp1testdashboardcomponenttest,
                columns: columns_dashapp1testdashboardcomponenttest,
                pageLength: 10,
                scrollX: true,
                searching: true,
                paging: true,
                info: true,
                ordering: true,
                order: order_dashapp1testdashboardcomponenttest,
            }
  
        
  
        var table_dashapp1testdashboardcomponenttest = $('#dashapp1testdashboardcomponenttest_table').DataTable(options);
    </script>
  
  
  
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs0-BasicTable]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  
  <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%">
      <thead>
      
      </thead>
      <tbody>
      
      </tbody>
    </table>
  </div>
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs0-Table]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%"></table>
  </div>
  
  
    
      <script id="data_dashapp1testdashboardcomponenttest" type="application/json">"value"</script>
    
    <script id="columns_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script id="order_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script type="module">
        var columns_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('columns_dashapp1testdashboardcomponenttest').textContent);
        var order_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('order_dashapp1testdashboardcomponenttest').textContent);
  
        
            let data_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('data_dashapp1testdashboardcomponenttest').textContent);
            let rows_dashapp1testdashboardcomponenttest = data_dashapp1testdashboardcomponenttest.data;
  
            var options = {
                data: rows_dashapp1testdashboardcomponenttest,
                columns: columns_dashapp1testdashboardcomponenttest,
                pageLength: 10,
                scrollX: true,
                searching: true,
                paging: true,
                info: true,
                ordering: true,
                order: order_dashapp1testdashboardcomponenttest,
            }
  
        
  
        var table_dashapp1testdashboardcomponenttest = $('#dashapp1testdashboardcomponenttest_table').DataTable(options);
    </script>
  
  
  
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs1-BasicTable]
  '''
  
  
  
      <div hx-get="/dash/app1/testdashboard/@component/test/"
           hx-trigger="intersect once delay:1ms">
          <div class="htmx-indicator">
              
  <div class="loading-img">
      <img src="/static/dashboards/loading.svg" />
  </div>
          </div>
      </div>
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs1-Table]
  '''
  
  
  
      <div hx-get="/dash/app1/testdashboard/@component/test/"
           hx-trigger="intersect once delay:1ms">
          <div class="htmx-indicator">
              
  <div class="loading-img">
      <img src="/static/dashboards/loading.svg" />
  </div>
          </div>
      </div>
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs2-BasicTable]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  
  <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%">
      <thead>
      
      </thead>
      <tbody>
      
      </tbody>
    </table>
  </div>
  
          </div>
      
  
  
  
  '''
# ---
# name: test_render[True-component_kwargs2-Table]
  '''
  
  
  
      
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              <div id="dashapp1testdashboardcomponenttest">
    <table id="dashapp1testdashboardcomponenttest_table" class="table" style="width:100%"></table>
  </div>
  
  
    
      <script id="data_dashapp1testdashboardcomponenttest" type="application/json">"value"</script>
    
    <script id="columns_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script id="order_dashapp1testdashboardcomponenttest" type="application/json">""</script>
    <script type="module">
        var columns_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('columns_dashapp1testdashboardcomponenttest').textContent);
        var order_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('order_dashapp1testdashboardcomponenttest').textContent);
  
        
            let data_dashapp1testdashboardcomponenttest = JSON.parse(document.getElementById('data_dashapp1testdashboardcomponenttest').textContent);
            let rows_dashapp1testdashboardcomponenttest = data_dashapp1testdashboardcomponenttest.data;
  
            var options = {
                data: rows_dashapp1testdashboardcomponenttest,
                columns: columns_dashapp1testdashboardcomponenttest,
                pageLength: 10,
                scrollX: true,
                searching: true,
                paging: true,
                info: true,
                ordering: true,
                order: order_dashapp1testdashboardcomponenttest,
            }
  
        
  
        var table_dashapp1testdashboardcomponenttest = $('#dashapp1testdashboardcomponenttest_table').DataTable(options);
    </script>
  
  
  
  
          </div>
      
  
  
  
  '''
# ---
